packages/astro/src/core/build/generate.ts

```diff
@@ -176,7 +176,11 @@ async function generatePath(
 
 	debug('build', `Generating: ${pathname}`);
 
-	const site = astroConfig.site;
+	// If a base path was provided, append it to the site URL. This ensures that
+	// all injected scripts and links are referenced relative to the site and subpath.
+	const site = astroConfig.base && astroConfig.base !== './'
+		? joinPaths(astroConfig.site?.toString() || 'http://localhost/', astroConfig.base)
+		: astroConfig.site;
```

Comment:
This is the fix I expected. Are we checking that `astroConfig.site` doesn't also end with `base`? Maybe that's validated when we load the config.

Reply:
Yep!  We already throw an error if the `site` includes a URL path

> [config] Astro found issue(s) with your configuration:
  ! site  "site" must be a valid URL origin (ex: "https://example.com") but cannot contain a URL path (ex: "https://example.com/blog"). Use "base" to configure your deployed URL path.